api/app/c_haines/severity_index.py

```diff
@@ -0,0 +1,453 @@
+        prediction_timestamp: datetime):
+    """ Check if we have a c-haines record """
+    result = get_c_haines_prediction(session, model_run, prediction_timestamp)
+    return result.count() > 0
+
+
+def get_prediction_date(model_run_timestamp, hour) -> datetime:
+    """ Construct the part of the filename that contains the model run date
+    """
+    return model_run_timestamp + timedelta(hours=hour)
+
+
+def model_prediction_hour_iterator(model: ModelEnum):
+    """ Return a prediction hour iterator.
+    Each model has a slightly different set of prediction hours. """
+    if model == ModelEnum.GDPS:
+        # GDPS goes out real far, but in 3 hour intervals.
+        for hour in range(0, 241, 3):
+            yield hour
+    elif model == ModelEnum.RDPS:
+        # RDPS goes out 3 1/2 days.
+        for hour in range(0, 85):
+            yield hour
+    elif model == ModelEnum.HRDPS:
+        # HRDPS goes out 2 days.
+        for hour in range(0, 49):
+            yield hour
+    else:
+        raise UnhandledPredictionModelType()
+
+
+def make_model_run_base_url(model: ModelEnum, model_run_start: str, forecast_hour: str):
+    """ Return the base url for the grib file.
+    The location of the files differs slightly for each model. """
+    if model == ModelEnum.GDPS:
+        return 'https://dd.weather.gc.ca/model_gem_global/15km/grib2/lat_lon/{HH}/{hhh}/'.format(
+            HH=model_run_start, hhh=forecast_hour)
+    if model == ModelEnum.RDPS:
+        return 'https://dd.weather.gc.ca/model_gem_regional/10km/grib2/{HH}/{hhh}/'.format(
+            HH=model_run_start, hhh=forecast_hour
+        )
+    if model == ModelEnum.HRDPS:
+        return 'https://dd.weather.gc.ca/model_hrdps/continental/grib2/{HH}/{hhh}/'.format(
+            HH=model_run_start, hhh=forecast_hour)
+    raise UnhandledPredictionModelType()
+
+
+def make_model_run_filename(
+        model: ModelEnum, level: str, date: str, model_run_start: str, forecast_hour: str):
+    """ Return the filename of the grib file.
+    The filename for each model differs slightly. """
+    if model == ModelEnum.GDPS:
+        return 'CMC_glb_{}_latlon.15x.15_{}{HH}_P{hhh}.grib2'.format(
+            level, date, HH=model_run_start, hhh=forecast_hour)
+    if model == ModelEnum.RDPS:
+        return 'CMC_reg_{}_ps10km_{}{HH}_P{hhh}.grib2'.format(
+            level, date, HH=model_run_start, hhh=forecast_hour)
+    if model == ModelEnum.HRDPS:
+        return 'CMC_hrdps_continental_{level}_ps2.5km_{date}{HH}_P{hhh}-00.grib2'.format(
+            level=level, date=date, HH=model_run_start, hhh=forecast_hour)
+    raise UnhandledPredictionModelType()
+
+
+def make_model_levels(model: ModelEnum):
+    """ Return list of layers. (The layers are named slightly differently for HRDPS) """
+    if model == ModelEnum.HRDPS:
+        return ['TMP_ISBL_0700', 'TMP_ISBL_0850', 'DEPR_ISBL_0850']
+    return ['TMP_ISBL_700', 'TMP_ISBL_850', 'DEPR_ISBL_850']
```

Comment:
could you document what variables these are referring to? (specifically, I'm not sure what DEPR is)